processor/resourcedetectionprocessor/resourcedetection_processor.go

```diff
@@ -17,6 +17,7 @@ import (
 )
 
 type resourceDetectionProcessor struct {
+	config             *Config
```

Comment:
See comment above, can we just pass the MiddlewareID instead of the whole Config?